core/tests/bank_forks.rs

```diff
@@ -38,400 +39,455 @@ mod tests {
         genesis_config_info: GenesisConfigInfo,
     }
 
-    fn setup_snapshot_test(snapshot_interval_slots: u64) -> SnapshotTestConfig {
-        let accounts_dir = TempDir::new().unwrap();
-        let snapshot_dir = TempDir::new().unwrap();
-        let snapshot_output_path = TempDir::new().unwrap();
-        let genesis_config_info = create_genesis_config(10_000);
-        let bank0 = Bank::new_with_paths(
-            &genesis_config_info.genesis_config,
-            vec![accounts_dir.path().to_path_buf()],
-            &[],
-        );
-        bank0.freeze();
-        let mut bank_forks = BankForks::new(bank0);
-        bank_forks.accounts_hash_interval_slots = snapshot_interval_slots;
-
-        let snapshot_config = SnapshotConfig {
-            snapshot_interval_slots,
-            snapshot_package_output_path: PathBuf::from(snapshot_output_path.path()),
-            snapshot_path: PathBuf::from(snapshot_dir.path()),
-            compression: CompressionType::Bzip2,
-        };
-        bank_forks.set_snapshot_config(Some(snapshot_config.clone()));
-        SnapshotTestConfig {
-            accounts_dir,
-            snapshot_dir,
-            _snapshot_output_path: snapshot_output_path,
-            snapshot_config,
-            bank_forks,
-            genesis_config_info,
-        }
-    }
+    macro_rules! DEFINE_SNAPSHOT_VERSION_PARAMETERIZED_TEST_FUNCTIONS {
+        ($x:ident) => {
+            #[allow(non_snake_case)]
+            mod $x {
+                use super::*;
+
+                const SNAPSHOT_VERSION: SnapshotVersion = SnapshotVersion::$x;
```

Comment:
@svenski123 Could you try to minimize the code put under `macro_rules!`?  I know this is the straight-forward way to do these tests with the unfortunate lack of property-based testing with cargo test facility. So, I understand your intention. :)

But, I think minimizing the macro code is doable and desirable for following reasons:

- Reduces diff (This is important metrics, considering possible back-ports)
- Generally macro code is hard to develop because of indirect compilation messages.

My suggestion is like this if you want to keep nice test result output(https://buildkite.com/solana-labs/solana/builds/26005#41e8769e-a92c-47b3-a621-e2969d08d70e/76-1810):

```rust
-    #[test]
-    fn test_bank_forks_snapshot_n() {
+    fn test_bank_forks_snapshot_n_inner(version) {
     }
     ...
     macro_rules! DEFINE_SNAPSHOT_VERSION_PARAMETERIZED_TEST_FUNCTIONS { 
         ($x:ident) => { 
             #[allow(non_snake_case)] 
             mod $x { 
                 use super::*; 
                 const SNAPSHOT_VERSION: SnapshotVersion = SnapshotVersion::$x;

                 #[test]
                 fn test_bank_forks_snapshot_n() {
                     test_bank_forks_snapshot_n_inner(SNAPSHOT_VERSION)
                 }
...
```

Or I'm fine even with this:

```rust
-#[test]
 fn test_bank_forks_snapshot_n_inner(version) {
 }

+#[test]
 fn test_bank_forks_snapshot_n() {
    test_bank_forks_snapshot_n_inner(V1_1_0)
    test_bank_forks_snapshot_n_inner(V1_2_0)
 }
...
```

Reply:
Refactored with only the stub functions in the macro